src/Compilers/Core/Portable/CommandLine/CommonCompiler.CompilerEmitStreamProvider.cs

```diff
@@ -20,16 +20,28 @@ private sealed class CompilerEmitStreamProvider : Compilation.EmitStreamProvider
             private readonly CommonCompiler _compiler;
             private readonly string _filePath;
             private Stream _streamToDispose;
+            private readonly Action<Exception, string> _exceptionHandler;
```

Comment:
>private readonly Action<Exception, string> _exceptionHandler; [](start = 12, length = 61)

Again, this is too generic. It would be better to inline the logic of the lambdas to this class.